idl/github.com/uber/cadence/shared.thrift

```diff
@@ -1230,10 +1230,20 @@ struct DescribeTaskListRequest {
   10: optional string domain
   20: optional TaskList taskList
   30: optional TaskListType taskListType
+  40: optional bool isAdmin
 }
 
 struct DescribeTaskListResponse {
   10: optional list<PollerInfo> pollers
+  20: optional i64 (js.type = "Long") backlogCountHint
```

Comment:
Let's create a type that holds all the detailed information. That way we have a clear distinction between what is set when detailed=true and what is not set when detailed=false. This also lets us add fields directly to DescribeTaskListResponse and have it still be clear which fields are for detailed and which fields are not for detailed.